### PR TITLE
[clang][Interp] Implement ComplexToReal casts

### PR DESCRIPTION
**Pull request by tbaederr**
Add a new emitComplexReal() helper function and use that for the new casts as well as the old __real implementation.

### CODE REVIEW

File: clang/test/AST/Interp/complex.cpp [hunk 1]
```diff
@@ -37,6 +37,18 @@ constexpr _Complex int I2 = {};
 static_assert(__real(I2) == 0, "");
 static_assert(__imag(I2) == 0, "");
 
+constexpr int ignoredCast() {
+  I2;
+  (int)I2;
+  /* (float)I2; FIXME*/
+  D1;
+  /* (int)D1; FIXME*/
```
Comment (tbaederr) [inline clang/test/AST/Interp/complex.cpp:45]: The commented-out tests need https://github.com/llvm/llvm-project/pull/77295 first.

**Comment by tbaederr** [comment]
Ping

File: clang/test/AST/Interp/complex.cpp [hunk 2]
```diff
@@ -37,6 +37,18 @@ constexpr _Complex int I2 = {};
...
+  I2;
+  (int)I2;
+  (float)I2;
+  D1;
+  (int)D1;
+  (double)D1;
+  return 0;
+}
+static_assert(ignoredCast() == 0, "");
+static_assert((int)I1 == 1, "");
```
Comment (AaronBallman) [inline clang/test/AST/Interp/complex.cpp:50]: Can you also add:
```
static_assert((float)D == 1.0f, "");
```
to show that we get the correct value out of the floating complex as well?

**Review by AaronBallman — APPROVED** [review]
LGTM with a testing request

**Pull request merged**